server/src/weaverbird/backends/mongo_translator/steps/convert.py

```diff
@@ -0,0 +1,87 @@
+from typing import List, Optional, Union
+
+from weaverbird.backends.mongo_translator.steps.types import MongoStep
+from weaverbird.pipeline.steps import ConvertStep
+
+TYPE_MAP = {
+    'boolean': 'bool',
+    'date': 'date',
+    'float': 'double',
+    'integer': 'long',  # better to cast into 64-bit integers. 32-bit integers ('int') cannot be casted to dates
+    'text': 'string',
+}
+
+
+def translate_convert(step: ConvertStep) -> List[MongoStep]:
+    mongo_add_fields = {}
+    mongo_type = TYPE_MAP[step.data_type] if step.data_type in TYPE_MAP else ''
+    for column in step.columns:
+        _input: Union[dict, str]
+        _on_null: Optional[str] = None
+
+        if mongo_type == 'date':
+            _input = {
+                '$switch': {
+                    'branches': [
+                        # Mongo cannot manage string representing only a year (e.g: "2020")
+                        # so we need to concatenate the month and day ("-01-01"):
+                        {
+                            'case': {
+                                '$and': [
+                                    {'$eq': [{'$type': f'${column}'}, 'string']},
+                                    {'$eq': [{'$strLenCP': f'${column}'}, 4]},
+                                ]
+                            },
+                            'then': {'$concat': [f'${column}', '-01-01']},
+                        },
+                        # Mongo cannot cast integers into date but only long into date,
+                        # so we manage the cast from int to long when needed:
+                        {
+                            'case': {'$eq': [{'$type': f'${column}'}, 'int']},
+                            'then': {'$toLong': f'${column}'},
+                        },
+                    ],
+                    'default': f'${column}',
+                }
+            }
```

Comment:
Since last version, we use `todate` step instead of the `convert` step when someone tries to cast a column to date types. This makes this branch less critical to implement exactly right.